docs/contributing.rst

```diff
@@ -392,156 +353,160 @@ We are glad to accept any sort of documentation: function docstrings,
 reStructuredText documents (like this one), tutorials, etc. reStructuredText
 documents live in the source code repository under the ``docs/`` directory.
 
-You can edit the documentation using any text editor, and then generate the
-HTML output by typing ``make`` from the ``docs/`` directory. Alternatively,
-``make html`` may be used to generate the documentation **with** the example
-gallery (which takes quite some time). The resulting HTML files will be
-placed in ``_build/html/stable`` and are viewable in a web browser.
-
 
 Building the documentation
 --------------------------
 
-First, make sure you have :ref:`properly installed <install_bleeding_edge>`
-the development version.
-
-Building the documentation requires installing some additional packages::
+Building the documentation requires installing sphinx::
 
-    pip install sphinx sphinx-gallery numpydoc matplotlib Pillow pandas scikit-image
+    pip3 install sphinx
 
 To build the documentation, you need to be in the ``docs`` folder::
 
     cd docs
 
 In the vast majority of cases, you only need to generate the full web site::
 
+    sphinx-build -b html . build/html
     make html
 
 
 Guidelines for writing documentation
 ------------------------------------
 
-It is important to keep a good compromise between mathematical and algorithmic
-details, and give intuition to the reader on what the algorithm does.
-
-Basically, to elaborate on the above, it is best to always
-start with a small paragraph with a hand-waving explanation of what the
-method does to the data. Then, it is very helpful to point out why the feature is
-useful and when it should be used - the latter also including "big O"
-(:math:`O\left(g\left(n\right)\right)`) complexities of the algorithm, as opposed
-to just *rules of thumb*, as the latter can be very machine-dependent. If those
-complexities are not available, then rules of thumb may be provided instead.
+* When writing docstrings, follow the `NumPy` template::
```

Comment:
Maybe add a link to Numpydoc: https://numpydoc.readthedocs.io/en/latest/format.html